UtilityApps/src/dumpBfield.cpp

```diff
@@ -41,12 +41,43 @@ static int invoke_dump_B_field(int argc, char** argv ){
   std::stringstream sstr ;
   sstr << argv[2] << " " << argv[3] << " " << argv[4] << " " << argv[5] << " " << argv[6] << " " << argv[7] ;
   
-  float xRange , yRange , zRange , dx , dy, dz ;
-  sstr >> xRange >> yRange >> zRange >> dx >> dy >> dz;
+  std::string RangeX , RangeY , RangeZ;
+  float dx , dy, dz ;
 
-  xRange *= dd4hep::cm;
-  yRange *= dd4hep::cm;
-  zRange *= dd4hep::cm;
+  sstr >> RangeX >> RangeY >> RangeZ >> dx >> dy >> dz;
+
+  size_t colon_posX = RangeX.find(':');
+  size_t colon_posY = RangeY.find(':');
+  size_t colon_posZ = RangeZ.find(':');
+
+  float minX=0, maxX=0, minY=0, maxY=0, minZ=0, maxZ=0;
+
+  if( colon_posX == std::string::npos || colon_posY == std::string::npos || colon_posZ == std::string::npos ) {
```

Comment:
What about splitting this for each axis, then one could mix symmetric and asymetric limit setting?

Reply:
Sure.  I committed the change.